src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -105,62 +105,70 @@ bool ParserSubquery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 {
     ParserSelectWithUnionQuery select;
     ParserExplainQuery explain;
-
-    if (pos->type != TokenType::OpeningRoundBracket)
-        return false;
-    ++pos;
-
     ASTPtr result_node = nullptr;
+    ParserKeyword s_kql("KQL");
 
-    if (ASTPtr select_node; select.parse(pos, select_node, expected))
+    if (s_kql.ignore(pos, expected))
     {
-        result_node = std::move(select_node);
+        if (!ParserKQLTaleFunction().parse(pos, result_node, expected))
+            return false;
     }
```

Comment:
we don't want to change clickhouse syntax - I think it's possible to implement this as a table function

Reply:
this one of our use cases, we need a syntax to embed kql statment inside a SQL query, like:
`select * from kql(table|column)`

Reply:
I believe it's possible to implement this as a function, maybe not that trivial though - look as a primary example `view` function (https://github.com/ClickHouse/ClickHouse/blob/master/src/TableFunctions/TableFunctionView.h). Most likely you will need to extend ExpressionListParsers (https://github.com/ClickHouse/ClickHouse/blob/master/src/Parsers/ExpressionListParsers.cpp) as you need to parse arguments differently.